### ISSUE TITLE
 trouble with command autocompletions

### ISSUE THREAD

**Issue by InvisibleRasta** [fish-shell/fish-shell#9495]
fish  version 3.4.0
Linux gentoo 5.15.88-gento

I am having troubles with command autocompletions. When I type for example "emerge -a" and tab it spits out "find: ‘/var/db/repos/gentoo-zh’: No such file or directory"

 **emerge -a**_find: ‘/var/db/repos/gentoo-zh’: No such file or directory_

I tried using   sh -c 'env HOME=$(mktemp -d) fish' but I get the same result

https://asciinema.org/a/1Gpw29bKRKupw2bfMVuVsHSw3

**Comment by faho** [comment]
What happens here is that the completions read the portage config files, which tell them where the packages even are. That's all a big bowl of spaghetti (portage is, at this level, a bit of a mess and we would much appreciate it if it just had a command to print the available/installed packages, like other package managers - this is information it needs to find out anyway!).

Anyway, it never suppresses errors about non-existent paths you have configured. We should probably just silence stderr on the awkward `find` calls we do here.

----

As a sidenote: Fish 3.4.0 is two minor releases out of date and missing 7.5% of all commits ever made to fish. We would really prefer if you upgrade, in which case you would already get adjustments to the gentoo completions. Especially if you want to report a bug it's always a good idea to confirm it actually happens in the latest version - we do sometimes fix bugs, and then we make releases to get those bug fixes to people.

**Comment by InvisibleRasta** [comment]
i updated to the latest version available on portage, version 3.6. I confirm that i get the same result, not sure how i can remove this annoying thing.

**Comment by faho** [comment]
`funced __fish_print_portage_available_pkgs` and then add a `2>/dev/null` to the `find` invocation.

(after that `funcsave __fish_print_portage_available_pkgs`)

Probably also a good idea to silence other portage completion functions like __fish_print_portage_installed_pkgs - `find` will print all sorts of errors that we're not interested in.

**Comment by InvisibleRasta** [comment]
@faho look i stumbled on this article https://forums.gentoo.org/viewtopic-t-1115990-start-0.html

**Comment by InvisibleRasta** [comment]
Yes portage creates backup files and fish reads from the backup file
